Dockerfile

```diff
@@ -1,13 +1,31 @@
-FROM alpine
+FROM ruby:2.6-alpine
 
-RUN apk update
-RUN apk add git openssh-client curl jq
+RUN apk add \
+    curl \
+    docker \
+    git \
+    jq \
+    openssh-client
+RUN gem install bundler
 
-RUN adduser app -D
 WORKDIR /home/app
-COPY components.conf update-version update-version-wrapper ./
-RUN chmod +r components.conf
-RUN chmod +rx update-version update-version-wrapper
-USER app
+COPY \
+    Gemfile \
+    Gemfile.lock \
+    ./
+RUN mkdir -p vendor/bundle && bundle install --path vendor/bundle
 
-CMD ./update-version-wrapper
+COPY \
+    LICENSE.txt \
+    README.md \
+    Rakefile \
+    sync_images.rb \
+    sync_images_wrapper \
+    ./
+# COPY treats directories differently /shrug
+COPY \
+    spec \
+    ./spec
```

Comment:
I would consider using `.dockerignore` to filter (un)desired files - I think it would result in cleaner and more maintainable code.

Reply:
I think an allowlist is more explicit and less likely to drift than a denylist.

Reply:
This can be done with `.dockerignore` too, I still think it's the preferable mechanism.

Example of whitelist.
```
*
!README.md
```

THis also has the benefit, although probably not very important in this case, that it speeds up the build - files ignored by `.dockerignore` are extempt from docker build context (ie. not copied over).

Reply:
I don't think ~4 entries in .dockerignore (or the same ~6 args to `COPY` as are in the Dockerfile now) is a big win. I think an include list is more explicit and safer than an ignore list, and I think "include this file" is clearer than "don't ignore this file".

> In addition, it has the benefit that it speeds up the build as the files ignored by .dockerignore are exempt from docker build context (ie. not copied over)

Not sure I understand this. The allowlist `COPY` approach does not include extra files either.

Reply:
> Not sure I understand this. The allowlist COPY approach does not include extra files either.

Because of docker client/server architecture when docker build happens, docker client has to send over all the files in the context to the docker server. FIles that are excluded via `.dockerignore` won't be sent over to the docker server.